### PR TITLE
add context

### PR DESCRIPTION
**Pull request by jmank88**
https://smartcontract-it.atlassian.net/browse/BCF-2887

Requires:
- https://github.com/smartcontractkit/chainlink-common/pull/490

Supports:
- https://github.com/smartcontractkit/chainlink/pull/14011

### CODE REVIEW

**Comment by jmank88** [comment]
The linter is very confused. I tried updating to a new version of golangci-lint and using the regular one instead of our custom action, but it is still failing the same way :thinking:

File: go.mod [hunk 1]
```diff
@@ -28,11 +24,12 @@ require (
 	github.com/beorn7/perks v1.0.1 // indirect
 	github.com/bits-and-blooms/bitset v1.10.0 // indirect
 	github.com/btcsuite/btcd/btcec/v2 v2.3.2 // indirect
+	github.com/btcsuite/btcd/chaincfg/chainhash v1.0.3 // indirect
```
Comment (jmank88) [inline go.mod:27]: Including this line prevents the ambiguous package problem that the btcd replace was addressing.

File: go.mod [hunk 2]
```diff
@@ -2,24 +2,20 @@ module github.com/smartcontractkit/chainlink-automation
 
 go 1.21
 
-replace (
-	github.com/btcsuite/btcd => github.com/btcsuite/btcd v0.23.3
-	golang/github.com/gogo/protobuf => golang/github.com/gogo/protobuf v1.3.3
```
Comment (jmank88) [inline go.mod:7]: The protobuf module was unused.

File: go.mod [hunk 3]
```diff
@@ -46,26 +43,23 @@ require (
...
-	golang.org/x/sys v0.15.0 // indirect
+	golang.org/x/sync v0.6.0 // indirect
+	golang.org/x/sys v0.16.0 // indirect
 	golang.org/x/tools v0.15.0 // indirect
 	google.golang.org/protobuf v1.31.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	rsc.io/tmplfunc v0.0.3 // indirect
 )
-
-exclude golang/github.com/influxdata/influxdb v1.8.3
```
Comment (jmank88) [inline go.mod:71]: This is unused

File: .github/workflows/go.yml [hunk 4]
```diff
@@ -49,7 +49,7 @@ jobs:
         go-version-file: "go.mod"
 
     - name: Run Unit Tests
-      run: go test -v ./... -coverpkg=./... -coverprofile=coverage.txt
```
Comment (jmank88) [inline .github/workflows/go.yml:52]: Dropping `-v` prints only relevant logs, leading to a succinct summary of passing packages, with full logs only for failing tests.

**Review by infiloop2 — APPROVED** [review]

**Review by krehermann — APPROVED** [review]

**Pull request merged**